analyses/snv-callers/compare_snv_callers.Rmd

```diff
@@ -0,0 +1,484 @@
+forward with mutations called by Strelka2, Mutect2, and Lancet. And, due to it's 
+oversensitivity that VarDict's data can be dropped entirely. 
+
+#### Individual Caller's Reports 
+
+- [Lancet](./plots/lancet/lancet_both_report.html)
+- [MuTect2](./plots/mutect2/mutect2_both_report.html)
+- [Strelka2](./plots/strelka2/strelka2_both_report.html)
+- [Vardict](./plots/vardict/vardict_both_report.html)
+
+*Notable Findings:*  
+
+- [VarDict calls more mutations](#upset-graph) than any of the other callers. 
+Many of these have [very low ](#vaf-distributions) suggesting that some may be sequencing errors. 
+
+- Mutect2, Strelka2, and Lancet do have [plenty of mutations in common](#upset-graph). 
+
+### Outline of analyses completed:
+
+- [Upset graph](#upset-graph)  
+- [VAF of combinations of callers](#vaf-for-combination-sets-of-callers)
+- [Base changes by caller](#base-changes)
+- [VAF distributions by caller](#vaf-distributions)
+- [TMB distributions by caller](#tmb-distributions)
+- [VAF correlations between callers](#vaf-correlations)
+- [TMB correlations between callers](#tmb-correlations)
+
+#### Usage
+
+To run this from the command line, use:
+```
+Rscript -e "rmarkdown::render('analyses/snv-callers/consensus_snv_callers.Rmd', 
+                              clean = TRUE)"
+```
+
+_This assumes you are in the top directory of the repository._
+
+## Setup
+
+#### Packages and functions
+
+Read in set up script.
+
+```{r}
+if (!("ggupset" %in% installed.packages())) {
+  install.packages("ggupset", repos = "http://cran.us.r-project.org")
+}
+if (!("UpSetR" %in% installed.packages())) {
+  install.packages("UpSetR", repos = "http://cran.us.r-project.org")
+}
+if (!("GGally" %in% installed.packages())) {
+  install.packages("GGally", repos = "http://cran.us.r-project.org")
+}
+if (!("grid" %in% installed.packages())) {
+  install.packages("grid", repos = "http://cran.us.r-project.org")
+}
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# Import this for the recalculation of TMB for the consensus mutation calls
+source(file.path("util", "wrangle_functions.R"))
+```
+
+Set up a function for calculating the percent off mutations of a caller for a 
+given variable. 
+
+```{r}
+perc_mutation_by_caller <- function(df, var) {
+  # Summarizes a variable and calculates the percent of mutations for a caller
+  # for each category
+  #
+  # Args:
+  #   df: a data.frame with `caller` as a column.
+  #   var: a string identifying the column of the variable you need to summarize by.
+  #
+  # Returns:
+  #   a data.frame with a summarized percent of caller mutations calculated in
+  #   the `percent` column.
+
+  # Summarize by the number of times each category shows up
+  count_df <- df %>%
+    dplyr::group_by(caller, eval(parse(text = var))) %>%
+    dplyr::summarise(count = dplyr::n())
+
+  # Calculate sum totals for each caller
+  totals <- count_df %>%
+    dplyr::group_by(caller) %>%
+    dplyr::summarise(total = sum(count, na.rm = TRUE))
```

Comment:
You can simplify these with some use of count() and add_count(). I *think* the below does the same as these plus the next few lines.

```
count_df <- df %>%
    dplyr::count(caller, eval(parse(text = var)), name = "count") %>%
    dplyr::add_count(caller, wt = count, "total") %>%
    dplyr::mutate(percent = count/total)
```

Reply:
Will try to use this!